### ISSUE TITLE
[feature] Add option to delete necessary files / directories when running conan config install

### ISSUE THREAD

**Issue by jwillikers** [conan-io/conan#9885]
An issue that I've run into is that restructuring the structure of Conan profiles in a shared Conan configuration via Git can fail when switching a file to a directory of the same name or vice versa.
This pretty much fails without explanation, so the error message should probably be improved.
Perhaps it would be best to make this default behavior, which wouldn't require anything special to ensure scheduled runs of `conan config install` don't break. Otherwise, there could be an option like `--force` which would all overwriting any conflicting directories / files.

To reproduce this issue:

1. Create a git repository to share Conan configuration through `conan config install`.

```
mkdir conan-config
cd conan-config
mkdir profiles
mkdir -p profiles/debug
touch profiles/debug/address-sanitizer
git init
git add -A
git commit -m "Initial commit"
```

2.  Install the repository using `conan config install`.

```
conan config install .
```

3. Now re-organize such that a directory is now a file in the `conan-config` repository.

```
mv profiles/debug/address-sanitizer profiles/address-sanitizer
rmdir profiles/debug
touch profiles/debug
git add -A
git commit -m "Restructure profiles"
```

4. Re-run `conan config install` to update the profiles.

```
conan config install
Config install:  /var/home/jordan/Projects/test/conan-config
Copying file address-sanitizer to /var/home/jordan/.conan/profiles
Copying file debug to /var/home/jordan/.conan/profiles
ERROR: Failed conan config install:
```

Note the vague error message.

- [X] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by lasote** [comment]
Hi, I think I understand but do you have an error trace you can share? Sounds a bit like a corner case, right?

**Comment by jwillikers** [comment]
@lasote Sorry, I probably should have filed this using the bug template. I'll update the issue to describe the exact steps to make this happen. I don't think this is quite a _corner_ case, though, as I think this could reasonably happen under normal usage of the `conan config install` command, especially as things are better organized as they grow and evolve over time.

**Comment by jwillikers** [comment]
@lasote I've added steps to reproduce the issue including the error trace... not that it is much of an error trace.

**Comment by lasote** [comment]
Thanks, I see not that is not that weird and is totally unexpected. I'll check what can we improve.

**Comment by lasote** [comment]
A fix will be released in 1.43. 
It will remove the previous file if a directory is going to be copied and it will remove the directory if now a file is going to be copied.

**Comment by jwillikers** [comment]
Awesome, that's fantastic!
